### PR TITLE
PE-690: Weekly Executive 2021-10-08 (goerli)

### PR DESCRIPTION
**Pull request by olivdb**
# Description

# Contribution Checklist

- [x] PR title starts with `(PE-<TICKET_NUMBER>)`
- [x] Code approved
- [x] Tests approved
- [x] CI Tests pass

# Checklist

- [x] Every contract variable/method declared as public/external private/internal
- [x] Consider if this PR needs the `officeHours` modifier
- [x] Verify expiration (`4 days + 2 hours` monthly and `30 days` for the rest)
- [x] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [x] Validate all addresses used are in Goerli changelog or known
- [x] Notify any external teams affected by the spell so they have the opportunity to review
- [x] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [x] Ensure contract is verified on `Goerli` etherscan
- [x] Change test to use Goerli spell address and deploy timestamp
- [x] Keep `Goerli-DssSpell.sol` and `Goerli-DssSpell.t.sol` the same, but make a copy in `archive`
- [ ] `squash and merge` this PR

### CODE REVIEW

File: src/Goerli-DssSpell.sol [hunk 1]
```diff
@@ -31,31 +30,144 @@ contract DssSpellAction is DssAction {
...
+        uint256 line;
+
+        // Offboard BAT-A
+        // https://vote.makerdao.com/polling/QmWJfX8U?network=mainnet#vote-breakdown
+
+        (,,,line,) = DssVat(DssExecLib.vat()).ilks("BAT-A");
+        totalLineReduction += line;
+        DssExecLib.setIlkLiquidationPenalty("BAT-A", 0);
+        DssExecLib.removeIlkFromAutoLine("BAT-A");
+        DssExecLib.setIlkDebtCeiling("BAT-A", 0);
```
[the remark this answers is not in the document]
Comment (olivdb) [reply src/Goerli-DssSpell.sol:81]: As discussed, I'm leaning towards reducing the ilk DCs with `setIlkDebtCeiling` and updating the global DC manually with `vat.file` given that using `decreaseIlkDebtCeiling` would still require manually converting the `line` precision from `RAD` to DAI.

File: src/Goerli-DssSpell.sol [hunk 2]
```diff
@@ -18,11 +18,14 @@ pragma solidity 0.6.12;
 
 import "dss-exec-lib/DssExec.sol";
 import "dss-exec-lib/DssAction.sol";
+import "lib/dss-interfaces/src/dapp/DSTokenAbstract.sol";
+import "lib/dss-interfaces/src/dss/VatAbstract.sol";
```
Comment (naszam) [inline src/Goerli-DssSpell.sol:22]: nit: you can use multiple imports:
`import { DSTokenAbstract, VatAbstract } from "dss-interfaces/Interfaces.sol";`

File: src/Goerli-DssSpell.t.sol [hunk 3]
```diff
@@ -279,7 +279,7 @@ contract DssSpellTest is DSTest, DSMath {
         // Test for spell-specific parameters
         //
```
Comment (naszam) [inline src/Goerli-DssSpell.t.sol:280]: housekeeping: we can remove L179/180 as specific of a previous spell and we can leave a comment for future need (`// Insert spell-only addresses here`).

File: src/Goerli-DssSpell.sol (hunk 2 above)
Comment (naszam) [reply src/Goerli-DssSpell.sol:22]: Good point.

File: src/Goerli-DssSpell.t.sol [hunk 4]
```diff
@@ -406,17 +406,17 @@ contract DssSpellTest is DSTest, DSMath {
             calc_cut:     9900
```
Comment (naszam) [inline src/Goerli-DssSpell.t.sol:406]: We can remove L45/58 once the `dss-interfaces` is upgraded.
Also, we'd need to replace `DssVestLike` with `VestAbstract` in tests as well, to make them working with the right interface.

File: src/Goerli-DssSpell.sol [hunk 5]
```diff
@@ -31,31 +34,198 @@ contract DssSpellAction is DssAction {
...
+
+        // Offboard BAT-A
+        // https://vote.makerdao.com/polling/QmWJfX8U?network=mainnet#vote-breakdown
+
+        (,,,line,) = vat.ilks("BAT-A");
+        totalLineReduction = add(totalLineReduction, line);
+        DssExecLib.setIlkLiquidationPenalty("BAT-A", 0);
+        DssExecLib.removeIlkFromAutoLine("BAT-A");
+        DssExecLib.setIlkDebtCeiling("BAT-A", 0);
+        DssExecLib.linearInterpolation({
```
[the remark this answers is not in the document]
Comment (hexonaut) [reply src/Goerli-DssSpell.sol:141]: Fix is here if you want to review and redeploy: https://github.com/makerdao/dss-lerp/pull/10

File: src/Goerli-DssSpell.sol [hunk 6]
```diff
@@ -31,31 +42,197 @@ contract DssSpellAction is DssAction {
...
+    uint256 constant TARGET_UNIV2LINKETH_MAT = 300 * RAY / 100;
+
+    function add(uint x, uint y) internal pure returns (uint z) {
+        require((z = x + y) >= x, "ds-math-add-overflow");
+    }
+    function sub(uint x, uint y) internal pure returns (uint z) {
+        require((z = x - y) <= x, "ds-math-sub-underflow");
+    }
+
     // Turn on office hours
```
Comment (brianmcmichael) [inline src/Goerli-DssSpell.sol:76]: OfficeHours override function only needs to return false. Default is true.

File: src/Goerli-DssSpell.sol [hunk 7]
```diff
@@ -31,31 +42,197 @@ contract DssSpellAction is DssAction {
...
+        // // CES-001 - 1_223_552 DAI - 0x25307aB59Cd5d8b4E2C01218262Ddf6a89Ff86da
+        // // https://vote.makerdao.com/polling/QmbM8u7Q?network=mainnet#vote-breakdown
+        // DssExecLib.sendPaymentFromSurplusBuffer(CES_WALLET, 1_223_552);
+
+        //
+        // MKR vesting
+        //
+
+        TokenLike(DssExecLib.getChangelogAddress("MCD_GOV")).approve(MCD_VEST_MKR_TREASURY, 700 * WAD);
+        DssVestLike(MCD_VEST_MKR_TREASURY).file("cap", 700 * WAD / 365 days);
```
Comment (brianmcmichael) [inline src/Goerli-DssSpell.sol:96]: Let's just make this 1000/365 days to add a little headroom and imply that it shouldn't be directly correlated to the streams.

Comment (brianmcmichael) [reply src/Goerli-DssSpell.sol:96]: We will have to update this again as soon as we add another stream anyways.

File: src/Goerli-DssSpell.sol [hunk 8]
```diff
@@ -31,31 +42,197 @@ contract DssSpellAction is DssAction {
...
+    uint256 constant CURRENT_UNIV2AAVEETH_MAT = 165 * RAY / 100;
+    uint256 constant CURRENT_UNIV2LINKETH_MAT = 165 * RAY / 100;
+
+    uint256 constant TARGET_BAT_MAT          = 800 * RAY / 100;
+    uint256 constant TARGET_LRC_MAT          = 2600 * RAY / 100;
+    uint256 constant TARGET_ZRX_MAT          = 900 * RAY / 100;
+    uint256 constant TARGET_UNIV2AAVEETH_MAT = 400 * RAY / 100;
+    uint256 constant TARGET_UNIV2LINKETH_MAT = 300 * RAY / 100;
+
+    function add(uint x, uint y) internal pure returns (uint z) {
```
Comment (brianmcmichael) [inline src/Goerli-DssSpell.sol:69]: Minor nit, but we might want to prefix the internal functions `_add()` so they don't conflict with DS-Math and they are easier to identify in code.

**Review by godsflaw — APPROVED** [review]
```haskell
Running 19 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testBATOffboardings() (gas: 3077967)
[PASS] testFail_notScheduled() (gas: 12258)
[PASS] testZRXOffboardings() (gas: 3078033)
[PASS] testLRCOffboardings() (gas: 3078055)
[PASS] test_auth_in_sources() (gas: 5836321)
[PASS] test_use_eta() (gas: 338034)
[PASS] testOnTime() (gas: 3028984)
[PASS] testUNIV2LINKETHOffboardings() (gas: 3078209)
[PASS] test_nextCastTime() (gas: 400525)
[PASS] testFailTooEarly() (gas: 372902)
[PASS] test_bytecode_matches() (gas: 1990481)
[PASS] test_auth() (gas: 9576737)
[PASS] testSpellIsCast_GENERAL() (gas: 12050336)
[PASS] testFailWrongDay() (gas: 373035)
[PASS] testCastCost() (gas: 3030793)
[PASS] testVestMKR() (gas: 3106173)
[PASS] testNewChainlogValues() (gas: 3032563)
[PASS] testFailTooLate() (gas: 373347)
[PASS] testUNIV2AAVEETHOffboardings() (gas: 3078649)
```

**Pull request merged**